hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -3110,4 +3117,8 @@ public boolean isBucketFSOptimized(String volName, String buckName)
     }
     return false;
   }
+
+  public BucketLayout getBucketLayout() {
+    return BucketLayout.DEFAULT;
```

Comment:
@aryangupta1998  Its hard coded to DEFAULT/LEGACY. This is the reason for the test case failure `TestOzoneClientMultipartUploadWithFSO.testListPartsWithPartMarkerGreaterThanPartCount`. [Failed_Build](https://github.com/apache/ozone/pull/2561/checks?check_run_id=3722349964).

getBucketLayout() has to be determined from bucket for the functions that are common to non-FSO and FSO.

Following methods in KeyManagerImpl has common for non-FSO and FSO execution path. Please modify these calls by reading bucketLayout from OmBucketInfo.
KeyManagerImpl#checkAccess() -> getBucketLayout()
and
KeyManagerImpl#listParts() -> getBucketLayout()

Reply:
Done